balancer/rls/balancer.go

```diff
@@ -284,6 +286,14 @@ func (b *rlsBalancer) UpdateClientConnState(ccs balancer.ClientConnState) error
 	b.updateCh.Put(resumePickerUpdates{done: done})
 	b.stateMu.Unlock()
 	<-done
+
+	if resizeCache {
```

Comment:
Why not do this operation above, where we compute it, and remove the bool entirely?

Reply:
`cacheMu` needs to be grabbed before `stateMu`, which is why we can't do this operation up there. Added a comment for the same.